src/components/common/Drawer.js

```diff
@@ -22,57 +32,88 @@ class Drawer extends Component {
+      side: 'right',
+      animated: true,
       to: 'close',
     });
     this.props.navigator.pop();
   };
 
   render() {
+    const { user } = this.props;
+    const { role } = this.props;
     return (
-      <View style={styles.container}>
-        <Button
-          style={styles.button}
-          // textStyle={styles.text}
-          title={strings.changeRole}
-          onPress={this.changeRole}
-        />
-        <Button
-          style={styles.button}
-          // textStyle={styles.text}
-          title={strings.changeUser}
-          onPress={this.logout}
-        />
+      <View style={styles.containerWrapper}>
+        <View style={styles.topHalf}>
+          <Text style={TextStyles.fieldTitle}> {`${user.name}`} </Text>
+          {
+            {
+              Gather: (
+                <CustomButton
+                  icon={HistoryIcon}
+                  title={strings.history}
+                  textStyle={{ color: Colors.black }}
+                  onPress={this.changeRole}
+                />
+              ),
+              Weigh: (
+                <CustomButton
+                  icon={PocketIcon}
+                  title="weigh button"
+                  textStyle={{ color: Colors.black }}
+                  style={{ width: 120 }}
+                  onPress={this.changeRole}
+                />
+              ),
+              Bale: (
+                <CustomButton
+                  icon={BaleIcon}
+                  title="bale button"
+                  textStyle={{ color: Colors.black }}
+                  onPress={this.changeRole}
+                />
+              ),
+            }[role]
+          }
+        </View>
+        <View style={styles.bottomHalf}>
+          <CustomButton
+            title={strings.changeRole}
+            icon={ChangeRoleIcon}
+            textStyle={{ margin: 10, color: Colors.black }}
+            style={{ width: 130 }}
+            onPress={this.changeRole}
+          />
+          <CustomButton
+            title={strings.changeUser}
+            icon={UserIcon}
+            textStyle={{ margin: 10, color: Colors.black }}
+            style={{ width: 130 }}
+            onPress={this.logout}
+          />
+        </View>
       </View>
     );
   }
 }
 
-const styles = StyleSheet.create({
-  container: {
-    flex: 1,
-    justifyContent: 'center',
-    alignItems: 'center',
-    backgroundColor: Colors.white,
-  },
-  containerWrapper: {
-    flex: 1,
-  },
-  button: {
-    justifyContent: 'center',
-    height: 50,
-  },
-  textButton: {
-    fontSize: 20,
-  },
-});
-
-Drawer.propTypes = {};
+Drawer.propTypes = {
+  user: PropTypes.string,
+  role: PropTypes.string,
+  logout: PropTypes.func.isRequired,
+  changeRole: PropTypes.func.isRequired,
+  navigator: PropTypes.object.isRequired,
+};
 
-Drawer.defaultProps = {};
+Drawer.defaultProps = {
+  user: 'not assigned',
+  role: 'not assigned',
+};
```

Comment:
Did you use 'not assigned' for any special reason? Is better to use false instead.